var/spack/repos/builtin/packages/pmix/package.py

```diff
@@ -33,26 +34,35 @@ class Pmix(AutotoolsPackage):
     homepage = "https://pmix.org"
     url = "https://github.com/pmix/pmix/releases/download/v3.1.3/pmix-3.1.3.tar.bz2"
     git = "https://github.com/openpmix/openpmix.git"
+
     maintainers("rhc54")
 
     license("BSD-3-Clause-Open-MPI")
 
     # Branches 4.2 & 5.0 will also need submodules
     version("master", branch="master", submodules=True)
+    version("5.0.2", sha256="28227ff2ba925da2c3fece44502f23a91446017de0f5a58f5cea9370c514b83c")
     version("5.0.1", sha256="d4371792d4ba4c791e1010100b4bf9a65500ababaf5ff25d681f938527a67d4a")
     version("5.0.0", sha256="92a85c4946346816c297ac244fbaf4f723bba87fb7e4424a057c2dabd569928d")
+    version("4.2.9", sha256="6b11f4fd5c9d7f8e55fc6ebdee9af04b839f44d06044e58cea38c87c168784b3")
+    version("4.2.8", sha256="09b442878e233f3d7f11168e129b32e5c8573c3ab6aaa9f86cf2d59c31a43dc9")
+    version("4.2.7", sha256="ac9cf58a0bf01bfacd51d342100234f04c740ec14257e4492d1dd0207ff2a917")
     version("4.2.6", sha256="10b0d5a7fca70272e9427c677557578ac452cea02aeb00e30dec2116d20c3cd0")
     version("4.2.5", sha256="a89c2c5dc69715a4df1e76fdc4318299386c184623a1d0d5eb1fb062e14b0d2b")
     version("4.2.4", sha256="c4699543f2278d3a78bdac72b4b2da9cd92d11d18478d66522b8991764b021c8")
     version("4.2.3", sha256="c3d9d6885ae39c15627a86dc4718e050baf604acda71b8b9e2ee3b12ad5c2d2a")
     version("4.2.2", sha256="935b2f492e4bc409017f1425a83366aa72a7039605ea187c9fac7bb1371cd73c")
     version("4.2.1", sha256="3c992fa0d653b56e0e409bbaec9de8fc1b82c948364dbb28545442315ed2a179")
+    version("4.1.3", sha256="c96a12bb5e565867b27f526611182801ecc0cb9dcb0146b195e77ed511eef9dd")
```

Comment:
Does anybody need these old versions? If not I would just restrict to adding the latest 4.2 and 5.0 versions

Reply:
I saw some of these patch releases for older versions contain a backported security fix. For example: https://github.com/openpmix/openpmix/releases/tag/v4.1.3 It seems they may still be useful for some groups.

Not sure what the best practice is for this kind of thing. We could at least mark versions @:4.1 as deprecated. But is there a point where we just remove unsupported versions.

Reply:
FWIW: the security fix has nothing to do with PMIx. It was a bug in Slurm that we agreed to fix with a patch to PMIx that prevents them from doing the wrong thing. If I had known all the misunderstandings it would create, I would have probably rethought that decision - but that's water under the bridge.

From a PMIx perspective, I really only support v5.0 at this point - any problems with v4.2 just get referred to update as I no longer fix that series. Anything before v4.2 is unsupported.

Reply:
Okay, I will update and mark the previous versions as deprecated.